src/ir/analysis/derive_partial_eq_or_partial_ord.rs

```diff
@@ -264,89 +270,108 @@ impl<'ctx> MonotoneFramework for CannotDerivePartialEqOrPartialOrd<'ctx> {
                         return ConstrainResult::Same;
                     } else {
                         trace!("    union layout cannot derive PartialEq or PartialOrd");
-                        return self.insert(id);
+                        // This depends on the fact that can_trivially_derive_partialeq_or_partialord returns true 
+                        // only for arrays greater than 32.
+                        // TODO: But is there can be only arrays? 
+                        // Why inside of the can_trivially_derive_partialeq_or_partialord is map_or(false) ?
+                        return self.insert(id, Reason::ArrayTooLarge);
```

Comment:
`l.opaque().can_trivially_derive_partialeq_or_partialord()` contains a check if the array is large enough or not.

https://github.com/rust-lang-nursery/rust-bindgen/blob/30d6e0417a691da4f4959c8083328241fc02d1fd/src/ir/layout.rs#L142-L146

but it feels ugly, because it depends on the implementation of `l.opaque().can_trivially_derive_partialeq_or_partialord()`. Do you see a better way there?

As an aside:
Now, I realize that it is generally ok to derive `PartialEq` for the opaque, but what it means to derive `PartialOrd` for the opaque? Maybe we should split those...

Reply:
We could define `enum CanDerive { Yes, No(Reason) }` and have `CanTriviallyDerive...` return `No(ArrayTooLarge)` so that we aren't relying on implicit information here.

> Now, I realize that it is generally ok to derive PartialEq for the opaque, but what it means to derive PartialOrd for the opaque? Maybe we should split those...

Want to file a new issue to discuss this?

Reply:
> Want to file a new issue to discuss this? 

Done https://github.com/rust-lang-nursery/rust-bindgen/issues/1013 !